File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    
    const map = new Map();
    
    for(let i = 0; i < completion.length; i++){
        const value = map.get(completion[i]);
        if(value){
            map.set(completion[i], value + 1);
        } else {
            map.set(completion[i], 1);
        }
        
    }
    
    
    for(let i = 0; i < participant.length; i++){
        const candidate = participant[i];
        
        const isCompleted = map.get(candidate);
        
        if(isCompleted){
            map.set(candidate, isCompleted - 1);
        } else {
            return candidate;
        }
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    part = {}
    for p in participant:
        existing = part.get(p)
        if existing:
            part[p] = existing + 1
        else:
            part[p] = 1
    
    for c in completion:
        part[c] -= 1
        
        if part[c] == 0:
            del part[c]
        
    return list(part.keys())[0]
    
    
        